stately/src/otherNativeMain/kotlin/co/touchlab/stately/concurrency/Lock.kt

```diff
@@ -14,15 +26,34 @@ actual class Lock actual constructor() {
   }
 
   actual fun tryLock():Boolean {
-    return lock.compareAndSet(0, 1)
+    return when (lock.compareAndSwap(0, Thread.id)) {
+      Thread.id -> {
+        // We already have the lock, we are just re-entering.
+        reenterCount.increment()
+        true
+      }
+      0 -> {
+        // Another thread didn't have the lock so we just took it.
```

Comment:
do you need to increment the count here?

Reply:
Maybe you do, if you go with my suggestion for unlock

Reply:
Yeah, I don't right now. The reentry count only increases if there are additional entries after the initial one. This just makes it slightly simpler because I only change one thing per entry or exit of a lock.